### PR TITLE
Migrate to TS

### PR DESCRIPTION
**Pull request by maneetgoyal**

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -70,18 +70,18 @@ function sortSummaryDictionary(
...
-      data.forEach((d) => {
-        // Create project dictionary: Record<string, {name: string; value: number}>
-        const summaryDictionary = createSummaryDictionary(d.details);
-        // Update "summary" property of the array element
-        const summary = sortSummaryDictionary(summaryDictionary);
-        d.summary = summary;
-      });
-    }
+export function calculateSummary(data: CalendarHeatmapDatum[]): CalendarHeatmapDatum[] {
+  const output = data;
```
Comment (qur786) [inline src/utils.ts]:
```suggestion
  let output = data;
```

File: src/utils.ts [hunk 2]
```diff
@@ -70,18 +70,18 @@ function sortSummaryDictionary(
...
+  if (Array.isArray(data[0].summary) === false) {
+    data.map((d) => {
```
Comment (qur786) [inline src/utils.ts]:
```suggestion
    output = data.map((d) => {
```

**Comment by qur786** [comment]
Things to work upon (@qur786 )
- [x] Month overview is not rendering
- [ ] removal of year labels after tranisiton from globals to other overviews

**Comment by qur786** [comment]
CPT2205021252-1054x608

File: src/calendar-heatmap.tsx [hunk 3]
```diff
@@ -0,0 +1,1735 @@
...
+        item_size: 10,
+        label_padding: 40,
+        max_block_height: 20,
+        transition_duration: 500,
+        tooltip_width: 250,
+        tooltip_padding: 15,
+      },
+      in_transition: false,
+      // First entry for history should be 'global', so that even if the user enters the chart first time he will be able to go back till global overview
+      history: ['global', props.overview ?? 'year'],
```
Comment (maneetgoyal) [inline src/calendar-heatmap.tsx]: @qur786 refactor such that you do this only when `props.overivew` is not `global`

**Comment by qur786** [comment]
CPT2205021649-1054x608

File: src/calendar-heatmap.tsx [hunk 4]
```diff
@@ -738,7 +738,7 @@ export class CalendarHeatmap extends Component<
       );
 
     // Add month data items to the overview
-    this.items.selectAll('.item-block-month').exit().remove();
```
Comment (qur786) [inline src/calendar-heatmap.tsx]: with exit(), while pressing back button and coming back to month overview it is not rendering the element.

File: src/calendar-heatmap.tsx [hunk 5]
```diff
@@ -739,9 +738,10 @@ export class CalendarHeatmap extends Component<
       );
 
     // Add month data items to the overview
-    let item_block = this.items.selectAll('.item-block-month').data(month_data);
-    item_block.exit().remove();
-    item_block
+    this.items.selectAll('.item-block-month').exit().remove();
```
Comment (qur786) [inline src/calendar-heatmap.tsx]: With separate exit and remove it is causing the `selected elements` not to group. so `each` function is unable to run for the array.

**Pull request closed without merging**